### PR TITLE
MGMT-3490 Add flag to control log collector must-gather

### PR DESCRIPTION
**Pull request by oshercc**

### CODE REVIEW

**Comment by YuviGold** [comment]
Why?

**Comment by ErezAlster** [comment]
@yuv

**Comment by ErezAlster** [comment]
@YuviGold, we want to analyze all installation attempt snapshots for data analysis.

**Comment by YuviGold** [comment]
@ErezAlster 
We don't have enough storage to save each test-infra run with must-gather artifacts on Jenkins

**Comment by ErezAlster** [comment]
@oshercc

File: Makefile [hunk 1]
```diff
@@ -312,7 +312,7 @@ delete_all_virsh_resources: destroy_all_nodes delete_minikube kill_all_port_forw
 	skipper run $(SKIPPER_PARAMS) 'discovery-infra/delete_nodes.py -ns $(NAMESPACE) -a'
 
 _download_logs:
-	discovery-infra/download_logs.py $(REMOTE_SERVICE_URL) $(LOGS_DEST) --cluster-id $(CLUSTER_ID)
+	discovery-infra/download_logs.py $(REMOTE_SERVICE_URL) $(LOGS_DEST) --cluster-id $(CLUSTER_ID) --must-gather
```
Comment (YuviGold) [inline Makefile:315]: must-gather shouldn't always be provided. Only for `download_all_logs`

File: Jenkinsfile.download_all_logs [hunk 2]
```diff
@@ -0,0 +1,45 @@
+pipeline {
```
Comment (YuviGold) [inline Jenkinsfile.download_all_logs:1]: I don't think we should create another pipeline

**Comment by YuviGold** [comment]
> @oshercc, what about controller logs, installer gather?
> I think that the paremater should be something like: "take only metadata and events log"

IMO collecting only events and metadata is the right approach. It's very lightweight and it doesn't arise storage concerns.

File: Jenkinsfile.download_all_logs (hunk 2 above)
Comment (oshercc) [reply Jenkinsfile.download_all_logs:1]: there is a different pipeline, 
renamed the old one to download_failed_cluster_logs and added Jenkinsfile.download_all_logs

File: Makefile (hunk 1 above)
Comment (oshercc) [reply Makefile:315]: now download_logs  collects only the failed logs with must-gather
so this job is not supposed to change.

added a second pipeline that collects all logs that will not collect must-gather.

**Comment by ErezAlster** [comment]
>> @oshercc, what about controller logs, installer gather?
>> I think that the paremater should be something like: "take only metadata and events log"
>
>IMO collecting only events and metadata is the right approach. It's very lightweight and it doesn't arise storage concerns.

I agree. only saying that going to the white list approach is better than what not to do.

File: Jenkinsfile.download_all_logs (hunk 2 above)
Comment (YuviGold) [reply Jenkinsfile.download_all_logs:1]: Why do we need two pipelines? It can be the same one with different parameters.

File: Makefile (hunk 1 above)
Comment (YuviGold) [reply Makefile:315]: I would prefer a way to quickly download logs without must-gather.
What do you say about:

```makefile
_download_logs:
	discovery-infra/download_logs.py $(REMOTE_SERVICE_URL) $(LOGS_DEST) --cluster-id $(CLUSTER_ID) $(ADDITIONAL_PARAMS)

download_logs:
	skipper make $(SKIPPER_PARAMS) _download_logs $(ADDITIONAL_PARAMS)

download_all_logs:
	skipper make $(SKIPPER_PARAMS) _download_all_logs ADDITIONAL_PARAMS="'--download-all --must-gather  ${ADDITIONAL_PARAMS}'"
```

File: Jenkinsfile.download_logs [hunk 3]
```diff
@@ -3,6 +3,9 @@ pipeline {
 
     parameters {
         string(name: 'REMOTE_SERVICE_URL', defaultValue: 'https://api.openshift.com', description: 'Service URL')
+        string(name: 'DOWNLOAD_ALL', defaultValue: '--download-all', description: 'Collect all logs (not only from failed clusters)')
```
Comment (YuviGold) [inline Jenkinsfile.download_logs:6]: Change both `DOWNLOAD_ALL` and `MUST_GATHER` to boolean parameters.
If this parameter was ticked - add the matching flag.

Default `DOWNLOAD_ALL` is `false`, while `MUST_GATHER` is `true`

File: Makefile [hunk 4]
```diff
@@ -312,16 +312,16 @@ delete_all_virsh_resources: destroy_all_nodes delete_minikube kill_all_port_forw
...
 
 _download_logs:
-	discovery-infra/download_logs.py $(REMOTE_SERVICE_URL) $(LOGS_DEST) --cluster-id $(CLUSTER_ID)
+	discovery-infra/download_logs.py $(REMOTE_SERVICE_URL) $(LOGS_DEST) --cluster-id $(CLUSTER_ID) $(ADDITIONAL_PARAMS)
 
 download_logs:
-	skipper make $(SKIPPER_PARAMS) _download_logs
+	skipper make $(SKIPPER_PARAMS) _download_logs $(ADDITIONAL_PARAMS)
 
 _download_all_logs:
```
Comment (YuviGold) [inline Makefile:320]: You can delete `_download_all_logs` and call from `download_all_logs` to `_download_logs` directly.

**Review by YuviGold — APPROVED** [review]

**Review by YuviGold — APPROVED** [review]

**Comment by oshercc** [comment]
retest

**Review by YuviGold — APPROVED** [review]

**Pull request merged**